src/textureremix/TextureRemix.java

```diff
@@ -64,29 +72,92 @@ public static void main(String[] args) {
         TextureRemixGui.main(args);
     }
     
+    private static void LoadPersistedSavePath() {
+        File f = new File( "savepath.txt" );
+        if( f.exists() && f.isFile() ) {
+            try {
+                List<String> lines = Files.readAllLines(Paths.get(f.getAbsolutePath()), Charset.defaultCharset() );
+                if( lines.size() > 0 )
+                    savepath = lines.get(0);
+                else {
+                    System.out.println( "File "+f.getAbsolutePath()+" is empty" );
+                }
+            } catch( IOException io ) {
+                // ignore exception, savepath will be empty
+                System.out.println( "Error reading file "+f.getAbsolutePath()+": "+io.toString() );
+            }
+        } else {
+            System.out.println( "File savepath.txt not found" );
+        }
+    }
+    
+    private static void SavePersistedSavePath() {
+        try (Writer writer = new BufferedWriter(new OutputStreamWriter(
+            new FileOutputStream("savepath.txt"), Charset.defaultCharset()))) {
+            writer.write(savepath);
+        } catch( IOException io ) {
+        }
+    }
+    
     public static void setPath(String path) {
+        if( path.contains("Select a folder") ) return;
+        System.out.println( "Savepath set to "+path );
         savepath = path;
+        SavePersistedSavePath();
+    }
+    
+    public static String getFilenameFromPath( String file ) {
+        int i = file.lastIndexOf( "." );
+        if( i != -1 ) {
+            file = file.substring(0, i);
+        }
+        i = file.lastIndexOf("/");
+        if( i != -1 )
+            file = file.substring( i+1 );
+        i = file.lastIndexOf("\\");
+        if( i != -1 )
+            file = file.substring( i+1 );
+        return file;
+    }
+    
+    public static void ShowError( String msg, String headline ) {
+        JOptionPane.showMessageDialog(null, msg, headline, JOptionPane.ERROR_MESSAGE );
```

Comment:
Would it be possible to move these two functions to the Gui class? Or at least the lines that draw the JOptionPane? I'm not fond of the swing element class being used directly by the backend class.